napari/utils/tree/selection.py

```diff
@@ -0,0 +1,21 @@
+from typing import Iterable, Optional, TypeVar
+
+from ...utils.events import EventedSet
+
+_T = TypeVar("_T")
+
+
+class Selection(EventedSet[_T]):
+    def __init__(self, data: Iterable[_T] = (), current: Optional[_T] = None):
+        super().__init__(data=data)
+        self.events.add(current=None)
+        self._current = current
+
+    @property
+    def current(self) -> Optional[_T]:
+        return self._current
+
+    @current.setter
+    def current(self, index: Optional[_T]):
+        previous, self._current = self._current, index
```

Comment:
We probably want to add in an equality check to block event emission if the item is the same